design-docs/handling-encodings.md

```diff
@@ -0,0 +1,120 @@
+(file names and comments) in archives. It applies to tasks which need to read, write and transform text files, as well as tasks which create
+archives (zip files, tar files, etc.)
+
+## Use Cases
+
+### Content Encoding
+
+Any time we need to generate a text file, or to read, transform and write the contents of a text file, we need to know what 
+charset(s) to use.
+
+Common situations include:
+
+* Copying and filtering a file from one location to another.
+* Creating an archive (zip, tar, etc.).
+
+[GRADLE-1267](https://issues.gradle.org/browse/GRADLE-1267) describes just one instance of where this functionality is needed.
+
+### Metadata Encoding
+
+Metadata in zip files can be encoded with a variety of charsets. 
+
+POSIX-compliant tar files, on the other hand, only use UTF8 for metadata. But non-compliant implementations can use other
+charsets and Gradle should be able to read them. 
+
+Jar, war and ear files only use UTF8 for metadata. 
+
+In Gradle however, up to version 2.12 at least, the Zip, Jar, War and Ear tasks all have an `encoding` option added in 
+[Pull Request 499](https://github.com/gradle/gradle/pull/499) which allows specifying a metadata charset, which defaults to
+the platform default encoding. The name of this option is confusing, and the default value is not correct. Moreover, jar, war and ear
+metadata should always be encoded in UTF8 instead of the platform default encoding. The Tar task should use UTF8, but it
+uses ASCII.
+
+## Solution
+
+### Content encoding
+
+This design generally proposes that any task which needs to read and write text should specify a charset used to encode/decode 
+characters into bytes.
+
+Notes:
+
+ - reading and writing characters isn't limited to files. It also applies to archive entries, network communication, etc.
+ - some tasks might need to use a charset to write that is different from the charset used to read. For the most frequent case,
+   i.e. copying and filtering files, it's a reasonable assumption to expect the developer to choose the target encoding for the
+   source files, and thus to use a single filtering charset both for reading and writing. By the way, when not filtering the 
+   files, the Copy task treats them as binary files, and thus uses the same encoding for the source and target files.
+
+### Metadata encoding
+
+The Zip task, which produces zip files, should allow specifying a metadata charset used to encode/decode the metadata
+of the archive. This should default to UTF8. 
+
+Jar, War, Ear and Tar tasks should not allow specifying a metadata charset, or at the very least should reject any metadata 
+charset other than UTF8.
+
+When reading from zip files using `ZipFileTree`, we should support a metadata charset option as well.
+
+When reading from tar files using `TarFileTree`, we should support a metadata charset option so that we can handle reading 
+tar files which were created by tools other than Gradle.
```

Comment:
It's more about reading **non-POSIX** tar files. Like the tar files created with current and older versions of Gradle, or with other tools.